src/main/java/de/fraunhofer/aisec/cpg/frontends/java/DeclarationHandler.java

```diff
@@ -210,15 +209,22 @@ public RecordDeclaration handleClassOrInterfaceDeclaration(
     // }
     name = getAbsoluteName(name);
 
-    List<Type> superTypes =
-        Stream.of(classInterDecl.getExtendedTypes(), classInterDecl.getImplementedTypes())
-            .flatMap(Collection::stream)
-            .map(this.lang::getTypeAsGoodAsPossible)
-            .collect(Collectors.toList());
-
     // add a type declaration
     RecordDeclaration recordDeclaration =
-        NodeBuilder.newRecordDeclaration(name, superTypes, "class", classInterDecl.toString());
+        NodeBuilder.newRecordDeclaration(name, "class", classInterDecl.toString());
+    recordDeclaration.setSuperClasses(
+        classInterDecl.getExtendedTypes().stream()
+            .map(this.lang::getTypeAsGoodAsPossible)
+            .collect(Collectors.toList()));
+    if (recordDeclaration.getSuperClasses().isEmpty()) {
+      List<Type> superClasses = new ArrayList<>();
+      superClasses.add(TypeParser.createFrom("java.lang.Object", true));
```

Comment:
Please do refer to class names as Strings but rather by `Object.class.getName()` to simplify refactoring.